include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -132,22 +310,58 @@ struct walk_n<_ExecutionPolicy, oneapi::dpl::__internal::__no_op>
 // walk_adjacent_difference
 //------------------------------------------------------------------------
 
-template <typename _ExecutionPolicy, typename _F>
-struct walk_adjacent_difference
+template <typename _ExecutionPolicy, typename _F, typename _Range1, typename _Range2>
+struct walk_adjacent_difference : public walk_vector_or_scalar_base<_Range1, _Range2>
 {
+  private:
+    using __base_t = walk_vector_or_scalar_base<_Range1, _Range2>;
     _F __f;
+    std::size_t __n;
 
-    template <typename _ItemId, typename _Acc1, typename _Acc2>
+  public:
+    walk_adjacent_difference(_F __f, std::size_t __n) : __f(__f), __n(__n) {}
+
+    template <typename _IsFull, typename _ItemId>
     void
-    operator()(const _ItemId __idx, const _Acc1& _acc_src, _Acc2& _acc_dst) const
+    __scalar_path_impl(_IsFull, const _ItemId __idx, const _Range1 __rng1, _Range2 __rng2) const
     {
-        using ::std::get;
-
         // just copy an element if it is the first one
         if (__idx == 0)
-            _acc_dst[__idx] = _acc_src[__idx];
+            __rng2[__idx] = __rng1[__idx];
+        else
+            __f(__rng1[__idx + (-1)], __rng1[__idx], __rng2[__idx]);
+    }
+    template <typename _IsFull, typename _ItemId>
+    void
+    __vector_path_impl(_IsFull __is_full, const _ItemId __idx, const _Range1 __rng1, _Range2 __rng2) const
+    {
+        using _ValueType = oneapi::dpl::__internal::__value_t<_Range1>;
+        _ValueType __rng1_vector[__base_t::__preferred_vector_size + 1];
```

Comment:
It is adds a default constructor requirement for value type of a range....
(value type may not have a default constructor)

Reply:
Yes, please see this comment: https://github.com/uxlfoundation/oneDPL/pull/1976#issuecomment-2605891689

Until yesterday all of these were implemented with the `__lazy_ctor_storage` utility so the type would not need to be default constructible. Compiler support for vectorization is only enabled for fundamental types which we require to invoke the vector path, so I have dropped this support which results in simpler code.